src/puppetlabs/puppetdb/queue.clj

```diff
@@ -0,0 +1,199 @@
+(ns puppetlabs.puppetdb.queue
+  (:import [java.nio.charset StandardCharsets]
+           [java.io InputStreamReader BufferedReader InputStream]
+           [java.util TreeMap HashMap]
+           [java.nio.file Files LinkOption]
+           [java.nio.file.attribute FileAttribute])
+  (:require [clojure.string :as str :refer [re-quote-replacement]]
+            [puppetlabs.stockpile.queue :as stock]
+            [clj-time.coerce :as tcoerce]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.command.constants :as constants]
+            [metrics.timers :refer [timer time!]]
+            [metrics.counters :refer [inc!]]
+            [clj-time.core :as time]
+            [puppetlabs.kitchensink.core :as kitchensink]
+            [slingshot.slingshot :refer [throw+]]
+            [clojure.core.async :as async]
+            [clojure.core.async.impl.protocols :as async-protos]
+            [puppetlabs.puppetdb.nio :refer [get-path]]))
+
+(def metadata-command-names
+  (vals constants/command-names))
+
+(defn stream->json [^InputStream stream]
+  (try
+    (-> stream
+        (InputStreamReader. StandardCharsets/UTF_8)
+        BufferedReader.
+        (json/parse-stream true))
+    (catch Exception e
+      (throw+ {:kind ::parse-error} e "Error parsing command"))))
+
+(defn metadata-str [received command version certname]
+  (format "%d_%s_%d_%s.json"
+          (tcoerce/to-long received) command version certname))
+
+(defn- metadata-rx [valid-commands]
```

Comment:
Heh, well I'm sure you know who did it; I probably did that because if felt like internal machinery, but it's certainly not critical.